_topics/en/merkle-tree-vulnerabilities.md

```diff
@@ -0,0 +1,268 @@
+---
+title: Merkle tree vulnerabilities
+
+## Optional.  Shorter name to use for reference style links e.g., "foo"
+## will allow using the link [topic foo][].  Not case sensitive
+# shortname: foo
+
+## Optional.  An entry will be added to the topics index for each alias
+#title-aliases:
+#  - Foo
+
+## Required.  At least one category to which this topic belongs.  See
+## schema for options
+topic-categories:
+  - Security Problems
+
+## Optional.  Produces a Markdown link with either "[title][]" or
+## "[title](link)"
+primary_sources:
+  - title: "Weaknesses in Bitcoin's Merkle Root Construction"
+    link: https://lists.linuxfoundation.org/pipermail/bitcoin-dev/attachments/20190225/a27d8837/attachment-0001.pdf
+
+  - title: "CVE-2012-2459: block merkle calculation exploit"
+    link: https://bitcointalk.org/?topic=102395
+
+  - title: Leaf-Node weakness in Bitcoin Merkle Tree Design
+    link: https://bitslog.com/2018/06/09/leaf-node-weakness-in-bitcoin-merkle-tree-design/
+
+## Optional.  Each entry requires "title" and "url".  May also use "feature:
+## true" to bold entry and "date"
+optech_mentions:
+  - title: "Bitcoin Core #29412 checks for every known way to mutate a valid block into an invalid block"
+    url: /en/newsletters/2024/03/06/#bitcoin-core-29412
+
+  - title: Disclosure of invalidity caching vulnerability
+    url: /en/newsletters/2019/03/12/#bitcoin-core-vulnerability-disclosure
+
+  - title: "Bitcoin Core #27608 protects against an invalidity caching vulnerability"
+    url: /en/newsletters/2023/05/17/#bitcoin-core-27608
+
+  - title: "CVE-2017-12842 leaf-node weakness"
+    url: /en/newsletters/2018/12/28/#cve-2017-12842
+
+  - title: "CVE-2017-12842: discussion about minimum transaction size"
+    url: /en/newsletters/2020/05/27/#minimum-transaction-size-discussion
+
+  - title: "CVE-2017-12842: merkle tree fix proposed in consensus cleanup soft fork"
+    url:  /en/newsletters/2019/03/05/#merkle-tree-attacks
+
+  - title: "CVE-2017-12842: discussion of lowering minimum relayable transaction size"
+    url: /en/newsletters/2022/10/19/#minimum-relayable-transaction-size
+
+  - title: "CVE-2017-12842: Bitcoin Core PR Review Club discussion about lowering min relayable tx size"
+    url: /en/newsletters/2022/11/09/#bitcoin-core-pr-review-club
+
+## Optional.  Same format as "primary_sources" above
+see_also:
+  - title: CVEs
+    link: topic cves
+
+## Optional.  Force the display (true) or non-display (false) of stub
+## topic notice.  Default is to display if the page.content is below a
+## threshold word count
+#stub: false
+
+## Required.  Use Markdown formatting.  Only one paragraph.  No links allowed.
+## Should be less than 500 characters
+excerpt: >
+  **Merkle tree vulnerabilities** are a class of problems in the design
+  of the merkle tree used by the Bitcoin consensus protocol.
+
+---
+
+A merkle tree whose number of elements is a power of two can trivially
```

Comment:
Maybe linking to https://en.wikipedia.org/wiki/Merkle_tree although great description here already...